uaclient/entitlements/repo.py

```diff
@@ -95,11 +123,131 @@ def _perform_enable(self, silent: bool = False) -> bool:
         return True
 
     def _perform_disable(self, silent=False):
+        if self.purge and self.origin:
+            repo_origin_packages = apt.get_installed_packages_by_origin(
+                self.origin
+            )
+
+            if not self.purge_kernel_check(repo_origin_packages):
+                return False
+
+            packages_to_reinstall = []
+            packages_to_remove = []
+            for package in repo_origin_packages:
+                alternatives = apt.get_alternative_versions_for_package(
+                    package, exclude_origin=self.origin
+                )
+                if alternatives:
+                    # We can call max(List[Version]) but mypy doesn't know.
+                    # Or doesn't like it.
+                    packages_to_reinstall.append(
+                        (package, max(alternatives))  # type: ignore
+                    )
+                else:
+                    packages_to_remove.append(package)
+
+            if not self.prompt_for_purge(
+                packages_to_remove, packages_to_reinstall
+            ):
+                return False
+
         if hasattr(self, "remove_packages"):
             self.remove_packages()
         self.remove_apt_config(silent=silent)
+
+        if self.purge and self.origin:
+            self.execute_purge(packages_to_remove, packages_to_reinstall)
+        return True
+
+    def purge_kernel_check(self, package_list):
+        linux_image_versions = [
+            package.name[len("linux-image-") :]
+            for package in package_list
+            if re.match(RE_KERNEL_PKG, package.name)
+        ]
+        if linux_image_versions:
+            print(messages.PURGE_KERNEL_REMOVAL.format(service=self.title))
+            print(" ".join(linux_image_versions))
+
+            current_kernel = system.get_kernel_info().uname_release
+            if current_kernel in linux_image_versions:
+                print(
+                    messages.PURGE_CURRENT_KERNEL.format(
+                        kernel_version=current_kernel
+                    )
+                )
```

Comment:
Let's always print this because the user will always want to know
```suggestion
            print(
                messages.PURGE_CURRENT_KERNEL.format(
                    kernel_version=current_kernel
                )
            )
```